compiler/src/dmd/parse.d

```diff
@@ -4376,7 +4377,15 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
                     if (token.value == TOK.leftParenthesis)
                     {
                         nextToken();
-                        ealign = parseExpression();
+                        if (token.value == TOK.default_)
+                        {
+                            nextToken();
+                            ealign = null;
+                        }
+                        else
+                        {
+                            ealign = parseExpression();
```

Comment:
according to the spec should be `parseAssignExpression`

Reply:
Done by merging master after #20588 was merged.